src/checks/#cc4a#modern_language.clas.abap

```diff
@@ -0,0 +1,1157 @@
+    ELSE.
+      add_finding(
+      EXPORTING
+         procedure = procedure
+         statement_index = statement_index
+         code = c_code_line_exists
+         pseudo_comment = c_ps_line_exists
+      CHANGING findings = findings ).
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD analyze_loop.
+    DATA code_line_index TYPE string.
+    DATA code_line_exists TYPE string.
+    DATA code_lines TYPE if_ci_atc_quickfix=>ty_code.
+    DATA key  TYPE string.
+
+    DATA(statement) = procedure-statements[ statement_index ].
+    IF analyzer->find_clause_index(  tokens = statement-tokens clause = 'OR' ) <> 0
+    OR analyzer->find_clause_index(  tokens = statement-tokens clause = 'ASSIGNING' ) <> 0
+    OR analyzer->find_clause_index(  tokens = statement-tokens clause = 'INITIAL' ) <> 0
+    OR analyzer->find_clause_index(  tokens = statement-tokens clause = 'NOT' ) <> 0.
+      RETURN.
+    ENDIF.
+    DATA(key_idx) = analyzer->find_clause_index(  tokens = statement-tokens clause = 'WHERE' ).
+    IF key_idx = 0 OR statement-tokens[ key_idx + 1 ]-lexeme CP '(*'.
+      RETURN.
+    ENDIF.
+    DATA(table) = statement-tokens[ 3 ]-lexeme.
+    IF table CP '*('.
+      RETURN.
+    ENDIF.
+    DATA(result_idx) = analyzer->find_clause_index( tokens = statement-tokens clause = 'INTO' ).
+    IF result_idx <> 0.
+      ASSIGN statement-tokens[ result_idx + 1 ] TO FIELD-SYMBOL(<token>).
+      IF is_used( procedure = procedure from_index = statement_index + 1 full_name = <token>-references[ lines( <token>-references ) ]-full_name ).
+        RETURN.
+      ENDIF.
+    ENDIF.
+    IF analyzer->find_clause_index( tokens = statement-tokens start_index = key_idx + 1 clause = '(' ) <> 0
+    OR analyzer->find_clause_index( tokens = statement-tokens start_index = key_idx + 1 clause = ')' ) <> 0.
+      RETURN.
+    ENDIF.
+    DATA(token_idx) = key_idx + 1.
+    DO.
+      ASSIGN statement-tokens[ token_idx ] TO <token>.
+      append_token( EXPORTING token = <token> CHANGING result = key ).
+      token_idx += 1.
+      ASSIGN statement-tokens[ token_idx ] TO <token>.
+      CASE <token>-lexeme.
+        WHEN '=' OR 'EQ'.
+          key = |{ key } = |.
+        WHEN OTHERS.
+          RETURN.
+      ENDCASE.
+      DATA(and_idx) = analyzer->find_clause_index( tokens = statement-tokens start_index = token_idx + 1 clause = 'AND' ).
+      IF and_idx = 0.
+        append_tokens( EXPORTING tokens = statement-tokens from_idx = token_idx + 1 CHANGING result = key ).
+        EXIT.
+      ELSE.
+        append_tokens( EXPORTING tokens = statement-tokens from_idx = token_idx + 1 to_idx = and_idx - 1 CHANGING result = key ).
+        token_idx = and_idx + 1.
+      ENDIF.
+    ENDDO.
+    DATA(quickfixable) = abap_true.
+    LOOP AT procedure-statements FROM statement_index + 1 ASSIGNING FIELD-SYMBOL(<statement>).
+      DATA(tabix) = sy-tabix.
+      CASE <statement>-keyword.
+        WHEN 'ENDLOOP'.
+          DATA(end_idx) = sy-tabix.
```

Comment:
Declaring variable inside a scope and then later using it outside of it is confused, cf. also [style guide](https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#do-not-use-variables-outside-of-the-statement-block-they-are-declared-in)